### PR TITLE
Changed the way abbreviations are presented in MHV Secondary Nav

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Summary
See related issue for a more in-depth explanation of the changes. This PR removes the use of the `abbr` element and replaces it with the use of `aria-label`.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/84242

## Testing done
- Added unit tests
- Manual testing:
  - Load the MHV Landing page with the secondary nav enabled.
  - Use a screen reader to navigate the secondary nav items in mobile view and verify that `Appts` reads `Appointments` when using the screen reader.

## Screenshots
There were no changes to the UI.

## What areas of the site does it impact?
- MHV Secondary Navigation

## Acceptance criteria
- Remove the use of `abbr` which has inconsistent implementations in screen readers and voice control.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx [hunk 1]
```diff
@@ -25,28 +25,57 @@ describe('MHV Secondary Navigation Item Component', () => {
   });
 
   describe('handle abbreviations', () => {
-    it('when provided', () => {
+    it('when provided without arial label', () => {
+      const title = 'a title';
```
Comment (carlosfelix2) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx:29]: Note: This test stayed the same except for the last line.

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx [hunk 2]
```diff
@@ -25,28 +25,57 @@ describe('MHV Secondary Navigation Item Component', () => {
...
+          href="/my-health"
+        />,
+      );
+      expect(getAllByText(title).length).to.eql(1);
+      expect(getAllByText(abbr).length).to.eql(1);
+      expect(getAllByText(title)[0].parentNode.getAttribute('aria-label')).to.be
+        .null;
+    });
+
+    it('when provided with aria label', () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx:45]: Note: This is a new test. The PR diff is just off.

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx [hunk 3]
```diff
@@ -98,7 +127,7 @@ describe('MHV Secondary Navigation Item Component', () => {
 
   describe('reports custom events to GA', async () => {
     it('when a link is clicked', async () => {
-      const href = '/my-health/unit-test';
+      const href = '#/my-health/unit-test';
```
Comment (carlosfelix2) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx:130]: Note: There was an error being thrown to the console. Adding `#` is a workaround, so there is no navigation incurred, but the event is still fired.
`Error: Not implemented: navigation (except hash changes)`
Ref: https://stackoverflow.com/a/77617601/4334721

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx [hunk 4]
```diff
@@ -25,28 +25,57 @@ describe('MHV Secondary Navigation Item Component', () => {
...
           abbreviation={abbr}
+          ariaLabel={ariaLabel}
           icon="home"
           href="/my-health"
         />,
       );
       expect(getAllByText(title).length).to.eql(1);
       expect(getAllByText(abbr).length).to.eql(1);
+      expect(
+        getAllByText(title)[0].parentNode.getAttribute('aria-label'),
```
Comment (dcloud) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx:61]: Did you considered `getByLabelText('a label');` or `getByRole('link', { name: 'a label'})`? I believe those two methods are preferred for looking up elements by accessibility properties, with `getByRole` being the [highest priority query](https://testing-library.com/docs/queries/about#priority). 

[The docs for https://testing-library.com/docs/queries/bylabeltext](https://testing-library.com/docs/queries/bylabeltext) make it sound like its suited for aria-labels in addition to non-aria labels.

**Review by dcloud — COMMENTED** [review]
I can see the aria-label assuming the link name for Appointments. 

I think the tests could be tweaked to `getByRole` (or `getByLabelText`) which can query for things by accessible name. But I think you're fine to proceed without these changes

**Review by dcloud — DISMISSED** [review]
Sorry, intended that last review to be a 👍 , with minor comments

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx (hunk 4 above)
Comment (carlosfelix2) [reply src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx:61]: I did not know getByLabelText would do that. Let me try that out.

Comment (carlosfelix2) [reply src/platform/mhv/secondary-nav/tests/MhvSecondaryNavItem.unit.spec.jsx]: Updates tests

**Review by dcloud — APPROVED** [review]
Looks good, with the tweak to the tests

**Pull request merged**